addon-test-support/RouteInfoMock.ts

```diff
@@ -0,0 +1,75 @@
+interface RouteInfo {
+  readonly name: string;
+  readonly parent: RouteInfo | null;
+  readonly child: RouteInfo | null;
+  readonly localName: string;
+  readonly params: { [name: string]: unknown } | undefined;
+  readonly paramNames: string[];
+  readonly queryParams: { [name: string]: string };
+  readonly metadata: unknown;
+  find(
+    predicate: (this: any, routeInfo: RouteInfo, i: number) => boolean,
+    thisArg?: any
+  ): RouteInfo | undefined;
+}
```

Comment:
Here we should import the type from its current location in `@ember/routing`:

```suggestion
import type RouteInfo from '@ember/routing/-private/route-info';
```

That way we don't have to worry about keeping this in sync, other than to update the import location when we have a public API import for that type (soon!).